tests/gpflow/test_mean_functions.py

```diff
@@ -188,6 +190,42 @@ def test_linear_mean_functions_associative_property(mean_functions: Sequence[Mea
     assert_allclose(var_b, var_rhs)
 
 
+@pytest.mark.parametrize("batch", [(), (1, 2)])
+@pytest.mark.parametrize("degree", [0, 1, 3])
+@pytest.mark.parametrize("input_dim", [0, 1, 3])
+@pytest.mark.parametrize("output_dim", [1, 2])
+def test_polynomial__sanity(
+    batch: Tuple[int, ...], degree: int, input_dim: int, output_dim: int
+) -> None:
+    p = Polynomial(degree, input_dim, output_dim)
+    X = np.ones(batch + (input_dim,))
+    Y = p(X)
+    assert (batch) + (output_dim,) == Y.shape
+    assert_allclose(1.0, Y)  # Polynomial is initialised to constant 1.0.
+
+
+def test_polynomial__1d() -> None:
+    # Test on a 1D quadratic function, where we can easily work out the maths.
+    p = Polynomial(degree=2, w=[1.0, 2.0, 3.0])
+    X = np.array([[1.0], [2.0]])
+    Y = p(X)
+    assert_allclose(
+        [
+            [1.0 + 2.0 * 1.0 + 3.0 * (1.0 ** 2)],
+            [1.0 + 2.0 * 2.0 + 3.0 * (2.0 ** 2)],
+        ],
+        Y,
+    )
+
+
+def test_polynomial__linear() -> None:
+    # Test on a 3D linear function, where we can easily work out the maths.
+    p = Polynomial(degree=1, input_dim=3, w=[1.0, 2.0, 3.0, 4.0])
+    X = np.array([1.0, 2.0, 3.0])
```

Comment:
I'm surprised we can pass in a rank-1 tensor, should this not be `[[1.,2.,3.]]` (i.e. `1 x 3` shape) to represent a single 3D point?

Reply:
> I'm surprised we can pass in a rank-1 tensor, should this not be `[[1.,2.,3.]]` (i.e. `1 x 3` shape) to represent a single 3D point?

You can do either. If your input shape is `[1, 3]` then the output will have shape `[1, 1]`. If your input has shape `[3]`, then your output will have shape `[1]`. In general the input shape is `[batch..., input_dim]` where `batch...` can be whatever you want, including empty. :-)